core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java

```diff
@@ -88,12 +89,11 @@ public void initialize(BatchRuntimeContext context) throws Exception {
 
   @Override
   protected Map<String, String> getProperties() {
-    Map<String, String> properties;
-    properties = new HashMap<>(tableSinkConfig.getProperties().getProperties());
-
-    properties.put(Properties.BatchReadableWritable.NAME, tableSinkConfig.getName());
-    properties.put(Properties.BatchReadableWritable.TYPE, Table.class.getName());
-    return properties;
+    ExploreProperties.Builder builder = ExploreProperties.builder();
+    builder.addAll(new HashMap<>(tableSinkConfig.getProperties().getProperties()));
+    builder.setExploreTableName(tableSinkConfig.getName());
```

Comment:
we should be adding a new optional property to the table source and sink called 'exploreName'. It should default to the name of the Table, but users should be able to set it to something else if they want. You do this by adding a new variable to the table source and sink configs and annotating it as nullable.